src/Microsoft.Health.Fhir.Core/Messages/DataConvert/DataConvertRequest.cs

```diff
@@ -0,0 +1,39 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using MediatR;
+using Microsoft.Health.Fhir.Core.Features.Operations.DataConvert.Models;
+
+namespace Microsoft.Health.Fhir.Core.Messages.DataConvert
+{
+    public class DataConvertRequest : IRequest<DataConvertResponse>
+    {
+        public DataConvertRequest(string inputData, DataConvertInputDataType inputDataType, string registryServer, string templateCollectionReference, string entryPointTemplate)
+        {
+            EnsureArg.IsNotNullOrEmpty(inputData, nameof(inputData));
+            EnsureArg.IsNotNull<DataConvertInputDataType>(inputDataType, nameof(inputDataType));
+            EnsureArg.IsNotNull(registryServer, nameof(registryServer));
+            EnsureArg.IsNotNull(templateCollectionReference, nameof(templateCollectionReference));
+            EnsureArg.IsNotNullOrEmpty(entryPointTemplate, nameof(entryPointTemplate));
+
+            InputData = inputData;
+            InputDataType = inputDataType;
+            RegistryServer = registryServer;
+            TemplateCollectionReference = templateCollectionReference;
+            EntryPointTemplate = entryPointTemplate;
+        }
+
+        public string InputData { get; }
+
+        public DataConvertInputDataType InputDataType { get; }
+
+        public string RegistryServer { get; }
+
+        public string TemplateCollectionReference { get; }
+
+        public string EntryPointTemplate { get; }
```

Comment:
Can we throw some documentation on these properties?
I'm looking on test data:
```
        "name": "inputData",
            "valueString": "MSH|^~\\&|SIMHOSP|SFAC|RAPP|RFAC|20200508131015||ADT^A01|517|T|2.3|||AL||44|ASCII\nEVN|A01|20200508131015|||C005^Whittingham^Sylvia^^^Dr^^^DRNBR^PRSNL^^^ORGDR|\nPID|1|3735064194^^^SIMULATOR MRN^MRN|3735064194^^^SIMULATOR MRN^MRN~2021051528^^^NHSNBR^NHSNMBR||Kinmonth^Joanna^Chelsea^^Ms^^CURRENT||19870624000000|F|||89 Transaction House^Handmaiden Street^Wembley^^FV75 4GJ^GBR^HOME||020 3614 5541^HOME|||||||||C^White - Other^^^||||||||\nPD1|||FAMILY PRACTICE^^12345|\nPV1|1|I|OtherWard^MainRoom^Bed 183^Simulated Hospital^^BED^Main Building^4|28b|||C005^Whittingham^Sylvia^^^Dr^^^DRNBR^PRSNL^^^ORGDR|||CAR|||||||||16094728916771313876^^^^visitid||||||||||||||||||||||ARRIVED|||20200508131015||"
        },
        {
            "name": "templateCollectionReference",
            "valueString": "microsofthealth/fhirconverter:default"
        },
        {
            "name": "entryPointTemplate",
            "valueString": "ADT_A01"
        },
        {
            "name": "inputDataType",
            "valueString": "Hl7V2"
        }

```
Input data is a raw string which we need to convert, that part is clear.
InputData type is type of format in which raw string is encoded (I hope so)
TtemplateCollectionReference my guess this is related to output data format or maybe it's a converter entity. 
EntryPointTemplate I have no idea what is this out of reading code.

I'm sure it's all part of some standard or something like that, but I'm not sure how to find that standard, or documentation for it.

Reply:
I have added some explanation to these fields. Conversion is a [FHIR operation](https://www.hl7.org/fhir/resource-operation-convert.html) but it's not well defined yet and there is no standard doc for this. We define our own usage for legacy data conversion and will add a doc to this repo later on.